dkg/src/dkg_manager/mod.rs

```diff
@@ -117,60 +113,128 @@ impl<DKG: DKGTrait> DKGManager<DKG> {
     pub async fn run(
         mut self,
         in_progress_session: Option<DKGSessionState>,
-        dkg_start_event_rx: oneshot::Receiver<DKGStartEvent>,
+        mut dkg_start_event_rx: aptos_channel::Receiver<(), DKGStartEvent>,
         mut rpc_msg_rx: aptos_channel::Receiver<
             AccountAddress,
             (AccountAddress, IncomingRpcRequest),
         >,
         close_rx: oneshot::Receiver<oneshot::Sender<()>>,
     ) {
+        info!(
+            epoch = self.epoch_state.epoch,
+            my_addr = self.my_addr.to_hex().as_str(),
+            "[DKG] DKGManager started."
+        );
+        let mut interval = tokio::time::interval(Duration::from_millis(5000));
+
+        let (agg_trx_tx, mut agg_trx_rx) = aptos_channel::new(QueueStyle::KLAST, 1, None);
+        self.agg_trx_tx = Some(agg_trx_tx);
+
         if let Some(session_state) = in_progress_session {
             let DKGSessionState {
-                metadata,
                 start_time_us,
+                metadata,
                 ..
             } = session_state;
-            self.setup_deal_broadcast(start_time_us, &metadata)
-                .await
-                .expect("setup_deal_broadcast() should be infallible");
-        }
 
-        let (agg_trx_tx, mut agg_trx_rx) = aptos_channel::new(QueueStyle::KLAST, 1, None);
-        self.agg_trx_tx = Some(agg_trx_tx);
+            if metadata.dealer_epoch == self.epoch_state.epoch {
+                info!(
+                    epoch = self.epoch_state.epoch,
+                    "Found unfinished and current DKG session. Continuing it."
+                );
+                self.setup_deal_broadcast(start_time_us, &metadata)
+                    .await
+                    .expect("[DKG] setup_deal_broadcast() should be infallible");
```

Comment:
Maybe alert instead of expect?

Reply:
the only error is if inner state is not NotStarted?